### PR TITLE
Fixing different eclipse warnings (missing comments, unused parameters, static constants access) (#1486)

### PR DESCRIPTION
**Pull request by speckyspooky**
This change includes the fixing of different eclipse warnings:
- missing comments
- removed unused "Map param" from IPage-interface
- replaced access to constants in a more static way (based on eclipse warnings/hints)

### CODE REVIEW

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/layout/emitter/IPage.java [hunk 1]
```diff
@@ -50,11 +58,36 @@ public interface IPage {
...
+	 * @param imageData image data
+	 * @param extension image type
+	 * @param imageX    image x position
+	 * @param imageY    image y position
+	 * @param height    image height
+	 * @param width     image width
+	 * @param helpText  help text
+	 * @throws Exception handling exception
+	 */
 	void drawImage(String imageId, byte[] imageData, String extension, int imageX, int imageY, int height, int width,
```
Comment (wimjongman) [inline engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/layout/emitter/IPage.java:74]: Do you think we can break this API?

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/impl/TextAreaLayout.java [hunk 2]
```diff
@@ -195,10 +235,20 @@ public void newLine(boolean endParagraph) throws BirtException {
 		}
 	}
 
+	/**
+	 * Get the fre space
```
Comment (wimjongman) [inline engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/impl/TextAreaLayout.java:239]: Typo

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java [hunk 3]
```diff
@@ -16,10 +16,16 @@
 
 import java.awt.Color;
 
-import org.eclipse.birt.report.engine.content.IStyle;
+import org.eclipse.birt.report.engine.css.engine.value.css.CSSValueConstants;
 import org.eclipse.birt.report.engine.layout.pdf.font.FontInfo;
 import org.w3c.dom.css.CSSValue;
 
+/**
+ * Class to implements the text style
```
Comment (wimjongman) [inline engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:24]: Typo

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java [hunk 4]
```diff
@@ -62,98 +87,218 @@ public TextStyle(TextStyle style) {
...
+	 * Get the word spacing
+	 *
+	 * @return Return the word spacing
+	 */
 	public int getWordSpacing() {
 		return wordSpacing;
 	}
 
+	/**
+	 * Check if text is underlined
```
Comment (wimjongman) [inline engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:253]: the text

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java [hunk 5]
```diff
@@ -62,98 +87,218 @@ public TextStyle(TextStyle style) {
...
+	 *
+	 * @return Return the check result
```
Comment (wimjongman) [inline engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:255]: Wrong comment

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java [hunk 6]
```diff
@@ -62,98 +87,218 @@ public TextStyle(TextStyle style) {
...
+	 */
 	public boolean isUnderline() {
 		return underLine;
 	}
 
+	/**
+	 * Check if the text use line through
```
Comment (wimjongman) [inline engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:262]: uses line-through

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java [hunk 7]
```diff
@@ -62,98 +87,218 @@ public TextStyle(TextStyle style) {
...
+	 *
+	 * @return Return the check result
```
Comment (wimjongman) [inline engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:264]: Wrong doc

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java [hunk 8]
```diff
@@ -62,98 +87,218 @@ public TextStyle(TextStyle style) {
...
+	 */
 	public boolean isLinethrough() {
 		return lineThrough;
 	}
 
+	/**
+	 * Check if the text is underlined
```
Comment (wimjongman) [inline engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:271]: underlined or overlined?

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java [hunk 9]
```diff
@@ -62,98 +87,218 @@ public TextStyle(TextStyle style) {
...
+	 *
+	 * @return Return the check result
```
Comment (wimjongman) [inline engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:273]: check result?

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java [hunk 10]
```diff
@@ -62,98 +87,218 @@ public TextStyle(TextStyle style) {
...
+	 */
 	public boolean isOverline() {
 		return overLine;
 	}
 
+	/**
+	 * Get the text direction
+	 *
+	 * @return Return the text direction
```
Comment (wimjongman) [inline engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:282]: What are the valid values?

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java [hunk 11]
```diff
@@ -62,98 +87,218 @@ public TextStyle(TextStyle style) {
...
+	 */
 	public int getDirection() {
 		return direction;
 	}
 
+	/**
+	 * Check if the text has a hyperlink
+	 *
+	 * @return Return the check result
+	 */
```
Comment (wimjongman) [inline engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:292]: check result?

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java [hunk 12]
```diff
@@ -62,98 +87,218 @@ public TextStyle(TextStyle style) {
...
 	public boolean isHasHyperlink() {
 		return hasHyperlink;
 	}
 
+	/**
+	 * Set the text use a hyperlink
```
Comment (wimjongman) [inline engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:298]: non descriptive javadoc

**Review by wimjongman — CHANGES_REQUESTED** [review]

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/impl/TextAreaLayout.java (hunk 2 above)
Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/impl/TextAreaLayout.java:239]: will be done

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java (hunk 3 above)
Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:24]: sorry, which typo?

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java (hunk 4 above)
Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:253]: will be done

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java (hunk 5 above)
Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:255]: will be done

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java (hunk 3 above)
Comment (wimjongman) [reply engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:24]: remove the s from implements

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java (hunk 6 above)
Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:262]: will be done

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java (hunk 7 above)
Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:264]: will be done

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java (hunk 8 above)
Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:271]: will be changed to "overlined"

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java (hunk 9 above)
Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:273]: will be done

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java (hunk 10 above)
Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java:282]: I haven't checked all at all normaly RTL and LTR

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java (hunk 12 above)
Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java]: will be changed

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/layout/emitter/IPage.java (hunk 1 above)
Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/layout/emitter/IPage.java:74]: @wimjongman 
Of course a good question.
The point is that the parameter won't be used on the methods. But I have no problem if we decide to undo the change.

Your favorit would be to leave it like it is - correct?

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java (hunk 3 above)
Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/style/TextStyle.java]: changed

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/layout/emitter/IPage.java (hunk 1 above)
Comment (wimjongman) [reply engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/layout/emitter/IPage.java:74]: Yes. This is a public API. Other implementations could use the map.

Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/layout/emitter/IPage.java:76]: Ok, I have removed the changes of that.
You see it on the details and also on the changed files.

**Review by wimjongman — APPROVED** [review]
Thanks!

**Pull request merged**